cql3/functions/aggregate_fcts.cc

```diff
@@ -85,12 +101,14 @@ struct accumulator_for : public std::conditional_t<std::is_integral_v<T>,
 
 class impl_user_aggregate : public aggregate_function::aggregate {
     ::shared_ptr<scalar_function> _sfunc;
+    ::shared_ptr<scalar_function> _rfunc;
```

Comment:
Did you consider using the same approach for non-user aggregates? Defining a separate scalar function, instead of adding new methods for reduction? Maybe it's simpler overall.

Reply:
`impl_user_aggregate` also has added new `reduce()` method, which is calling this scalar function, so basically the approach is the same for native aggregates and UDAs. The difference is the UDA has defined reduction function as a scalar function and native aggregate has own implementation of reduce function.

Reply:
We should, in the future, make native aggregates have external state rather than internal accumulators. This will increase uniformity with UDAs.